### PR TITLE
Fellowships and Grants Page

### LINKED ISSUE #320: @/charterafrica/components/OpportunityCard
### Description

Reusing code (or knowledge) from Spotlight and Post card components, implement the `OpportunityCard` component to be used in grants and fellowship [sections](https://www.figma.com/file/gbnaO1FYPq16V2gAFbLOkl/CfA---EPD-Charter-Africa---Sep-2022?node-id=9108%3A28134&t=gHql57tAK6HPaLGe-0).

### PR DESCRIPTION
**Pull request by kelvinkipruto**
## Description

This PR implements creation of `grants and fellowship` page

Fixes #320 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Screenshots
screencapture-localhost-3000-opportunities-fellowships-2023-02-20-20_15_49

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kilemensi** [comment]
@koechkevin please review.

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js
Comment (koechkevin) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js:74]: Why not just use Grid ?

**Comment by koechkevin** [comment]
@kelvinkipruto this looks good.

Isn't this banner supposed to span the whole page on larger screens?

image

**Comment by kelvinkipruto** [comment]
> n't this banner supposed to span the whole page on larger screens?
@koechkevin Yes. Let me fix it

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js:18]: This technique was useful in MUI v4. In MUI v5, `sx` prop on a component handle most cases.

... also remember this technique is done **after** JS has been fully downloaded and executed on the client where `sx` is pure CSS.

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js:26]:
```suggestion
        rowSpacing={5}
```

... `Grid` [docs](https://mui.com/material-ui/react-grid/#row-amp-column-spacing)

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js:41]:
```suggestion
            href="/opportunities/grants"
```
... all of these are children of /opportunities & should also be passed in via props rather than hard-coded like this.

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js:56]:
```suggestion
      <Grid
        item
```
Everything inside a `Grid container` must be a `Grid item` (a `Grid item` can also be a `Grid container` at the same time of course i.e. `Grid item xs={12} container ...` or you can make the container child of the item i.e. `Grid item xs={12}><Grid container ...`). Please check the docs again.

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js:59]:
```suggestion
        direction={{ xs: "column", sm: "row }}
```

docs, docs, docs

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js:79]: ??

File: apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js:13]:
```suggestion
  if (!opportunity) {
    return null;
  }
  return (
```
basically if ... else , no need for space in between.

File: apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js:7]:
```suggestion
  const { title, description, deadline, link } = props;
```

Why the need to full opportunity object only to destructure it?

File: apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js:14]: Card needs to be clickable. See `Spotlight`

File: apps/charterafrica/src/components/GrantsFellowships/GrantFellowship.js
Comment (kilemensi) [inline apps/charterafrica/src/components/GrantsFellowships/GrantFellowship.js]: We have PageInfo component for this & it goes into a page, not another component. See `heldesk` page.

File: apps/charterafrica/src/components/GrantsFellowships/GrantFellowship.js
Comment (kilemensi) [inline apps/charterafrica/src/components/GrantsFellowships/GrantFellowship.js:95]: If same component is reusable for both grants and fellowships, then we need to find a common name for it... `GrantsAndFellowships`, `GrantsAndFellowshipsGrid`, `Opportunities`, `OpportunityGrid`, etc.

File: apps/charterafrica/src/components/Grants/Grants.js
Comment (kilemensi) [inline apps/charterafrica/src/components/Grants/Grants.js:31]:
```suggestion
```

File: apps/charterafrica/src/components/Grants/Grants.js
Comment (kilemensi) [inline apps/charterafrica/src/components/Grants/Grants.js]: I'm afraid to even ask what is going on here.

**Review by kilemensi — COMMENTED** [review]
👍🏽 @kelvinkipruto .

1. Please review the rest of charterafrica code. Most of the things have been done elsewhere but evidently you're not aware. This is why reviewing **every** PR is extremely important.
2. Please set out 30-45 minutes to read the MUI docs especially the sections around `sx` and `Grid`.

File: apps/charterafrica/src/pages/opportunities/[slug].page.js
Comment (koechkevin) [inline apps/charterafrica/src/pages/opportunities/[slug].page.js:28]: @kelvinkipruto Do we need to add Grid to the name? No.

File: apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js
Comment (koechkevin) [inline apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js:17]:
```suggestion

```

Comment (koechkevin) [reply apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js:17]: I think it is pretty clear

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js
Comment (koechkevin) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js:29]: Check the font weight on desktop

File: apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js
Comment (kilemensi) [inline apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js:24]:
```suggestion
      acc[status] = acc[status] || [];
      acc[status].push(grant)
```

File: apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js
Comment (kilemensi) [inline apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js:30]: Even this should come from the CMS via props (always think other languages).

File: apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js
Comment (kilemensi) [inline apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js:53]: For slow ones such as my self, why are we hiding them?

File: apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js
Comment (kilemensi) [inline apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js:79]: Isn't `neutral[200]` just `neutral.light`?

File: apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js
Comment (kilemensi) [inline apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js:83]:
```suggestion
                    my: 5,
```
If you need both top and bottom, use `my`; left and right, use `mx`.

File: apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.test.js
Comment (kilemensi) [inline apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.test.js:13]: Why would title be **Fellowships** and items be `grant`? I'm expecting either they'll also be named `fellowships` or if we want a generic name applicable to everything (grants, fellowships and events), `items` is usually a good start.

File: apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js:77]: CMS/props

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js:24]:
```suggestion
            color="neutral.dark"
```

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js:58]: props/CMS

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js
Comment (kilemensi) [inline apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js:16]:
```suggestion
  if (!grants?.length) {
```

**Review by kilemensi — COMMENTED** [review]
👍🏽 

--

1. Isn't `Funding` a better name than `GrandsFellowships`?
2. On mobile (haven't tested other screen sizes), cards don't align once Show All/Show Less is clicked.
Screenshot 2023-02-17 at 22-25-33 Grants   Fellowships charter AFRICA

3. On Desktop, there is a weird hover effect.
Screenshot 2023-02-17 at 22 22 44

4. On both mobile and desktop, image appears stretched as you can also see in the image above.

File: apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js
Comment (kelvinkipruto) [reply apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js:53]: On `mobile` and `tablet` screens, you show only one of the groups, i.e open call on grants, and technologies only on fellowships. This hides the rest of the groups and shows them on desktop

File: apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js
Comment (kelvinkipruto) [reply apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js:79]: I just picked this directly from the design. Is there an advantage of using `neutral.light` over `neutral[200]`? Or is it the extra import we have to make for the colors?

File: apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js
Comment (kelvinkipruto) [reply apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.js:83]: Awesome!!

File: apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.test.js
Comment (kelvinkipruto) [reply apps/charterafrica/src/components/GrantsAndFellowshipsGrid/GrantsAndFellowshipsGrid.test.js:13]: I agree. Items is a good one.

File: apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js
Comment (kelvinkipruto) [reply apps/charterafrica/src/components/OpportunityCardList/OpportunityCardList.js:29]: @koechkevin I can see that its `400` across all of the

**Comment by kilemensi** [comment]
@koechkevin , your review please

**Comment by kilemensi** [comment]
QQ @kelvinkipruto, I don't see events in here and https://github.com/CodeForAfrica/ui/issues/321 has been assigned to you. What's the plan?

**Comment by kelvinkipruto** [comment]
@kilemensi Waiting for review feedback on this. I want to work on #321 after merging this. I'm finishing up on #340 now.

**Review by koechkevin — APPROVED** [review]
LGTM.

Just fix erge conflicts

**Review by kilemensi — APPROVED** [review]
🚀 

---

There are a few things that we'll need to improve on e.g. show/hide logic is still a bit too difficult to follow, we'll probably use Payload [Globals](https://payloadcms.com/docs/configuration/globals) for app-wide localizable strings and not per component config, etc., but lets at least add Events and see how everything looks/works before making those calls.

**Pull request merged**
